design/ceph-external-cluster.md

```diff
@@ -0,0 +1,115 @@
+  - mds: for shared filesystem access
+
+## Rook Operator
+
+When the Rook operator is started, initially it is not aware of any clusters. When the admin creates the operator, he will want to configure the operator differently depending on if he wants to configure a local Rook cluster, or an external cluster.
+
+If external cluster management is required, the differences are:
+- The Rook Discover DaemonSet would not be necessary. Its purpose is to detect local devices, which is only needed for OSD configuration.
+- The Security Context Constraints (SCC) would not require all the privileges of a local cluster. These privileges are only required by mon and/or osd daemon pods, which are not running in the local cluster.
+  - `allowPrivilegedContainer`
+  - `allowHostDirVolumePlugin`
+  - `allowHostPID`
+  - `allowHostPorts`
+
+## CSI Driver
+
+The CSI driver is agnostic of whether Ceph is running locally or externally. The core requirement of the CSI driver is the list of mons and the keyring with which to connect. This metadata is required whether the cluster is local or external. The Rook operator will need to keep this metadata updated throughout the lifetime of the CSI driver.
+
+The CSI driver will be installed and configured by the Rook operator, similarly to any Rook cluster. The advantages of this approach instead of a standalone ceph-csi for external clusters include:
+- Provide a consistent experience across any Kubernetes/OpenShift deployment
+- Rook can install, configure, and update the Ceph CSI driver. Admins don't have to worry about the CSI driver.
+
+Question: How Rook would behave in the case where the admin deployed ceph-csi standalone as well as Rook? It seems reasonable not to support this, although it's not clear if there would actually be conflicts between the two.
+
+The flex driver would also be agnostic of the cluster for the same reasons, but we won’t need to worry about the flex driver going forward.
+
+## Rook-Ceph Cluster CRD
+
+In order for Rook to provide the storage to clients in the local cluster, the CephCluster CRD will be created in order for the operator to provide local management of the external cluster. There are several differences needed for the operator to be aware of an external cluster.
+
+1. Before the CephCluster CRD is created, some metadata must be initialized in local configmaps/secrets to allow the local cluster to manage the external cluster.
+   * mon endpoint(s) and admin keyring
+1. The mon, mgr, and osd daemons will not be managed by the local Rook operator. These daemons must be created and managed by the external cluster.
+1. The operator will make a “best effort” to keep the list of mons updated.
+   * If the mons change in the external cluster, the list of mons must be updated in the local cluster.
+   * The operator will need to query the Ceph status periodically (perhaps every minute). If there is a change to the mons, the operator will update the local configmaps/secrets.
```

Comment:
Rook should have a good default out of the box, but it might also make sense for this time period to be configurable in the CephCluster CRD.
Could Rook also expose a "status" item that provides a last-mon-check time and success/failure flag? This might be very helpful in debugging if Rook can't reach any mons.

Reply:
Agreed, would be helpful to have the status on the crd.